internal/ignition/ignition.go

```diff
@@ -78,7 +81,7 @@ func NewGenerator(workDir string, installerDir string, cluster *common.Cluster,
 // UploadToS3 uploads generated ignition and related files to the configured
 // S3-compatible storage
 func (g *installerGenerator) UploadToS3(ctx context.Context) error {
-	return uploadToS3(ctx, g.workDir, g.cluster.ID.String(), g.s3Client, g.log)
+	return uploadToS3(ctx, g.workDir, g.cluster, g.s3Client, g.log)
```

Comment:
What about disconnected environment, i don't think that we have s3 there. (on-prem)

Reply:
That's a good question. This doesn't look conditional currently so I'll just look into how we're doing it for the ignitions we create now in a disconnected env.

Reply:
Right, only [`job.go`](https://github.com/openshift/assisted-service/blob/da122b778fe322647f3660fe18cd1daac2c75f21/pkg/job/job.go#L343) uploads the files. [`local_job.go`](https://github.com/openshift/assisted-service/blob/da122b778fe322647f3660fe18cd1daac2c75f21/pkg/job/local_job.go#L28-L60) does not.

That said, I don't know how they're getting access to these files after generating them, but that problem must already be solved for the existing ignition configs, right?

Reply:
@mhrivnak can you weigh in here and ensure I'm not breaking on-prem stuff?

Reply:
I agree that this does not appear to affect the on-premise implementation.